uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/BAO/LeaveRequest.php

```diff
@@ -411,9 +412,15 @@ private static function validateLeaveDatesDoesNotOverlapContracts($params) {
       'end_date' => $toDate->format('Y-m-d'),
     ]);
 
-    if ($contract['count'] > 1) {
+    $contractForEndDate = civicrm_api3('HRJobContract', 'getcontractswithdetailsinperiod', [
```

Comment:
This variable name sounds like it's always a singular contract, it might be renamed something like `$contractsContainingToDate`. Also, is it possible that the $toDate is in the past and this call would return an inactive contract from the past, thus passing validation here?